### PR TITLE
Issue #3059 Update go version to 1.11.2

### LINKED ISSUE #3059: Support Go v1.11.2 in CI
We need to fix the `gofmt` errors. In centos ci with go version 1.11.2 there's some gofmt errors, that are preventing the integration test from running.
```
07:13:16 + make prerelease synopsis_docs link_check_docs
07:13:16 rm -rf /home/minishift_ci/gopath/pkg/linux_amd64/github.com/minishift
07:13:16 rm -rf out
07:13:16 rm -rf release
07:13:16 rm -f  docs/source/_tmp/*.md
07:13:16 cmd/minishift/cmd/config/config_test.go
07:13:16 cmd/minishift/cmd/config/view_test.go
07:13:16 pkg/minishift/docker/image/oci_image_handler.go
07:13:17 pkg/util/crypto.go
07:13:17 make: *** [fmtcheck] Error 1
```
-----------------------------------
```
07:13:14 + go version
07:13:14 go version go1.11.2 linux/amd64
```

centos ci console: https://ci.centos.org/job/minishift-pr/5484/console

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #3059

### CODE REVIEW

**Comment by anjannath** [comment]
@LalatenduMohanty It is go version v1.11.2

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
     macos:
       xcode: "9.0"
     environment:
-        GOVERSION: 1.10.2
+        GOVERSION: 1.11.2
```
Comment (praveenkumar) [inline .circleci/config.yml]: :+1:

File: cmd/minishift/cmd/config/config_test.go [hunk 2]
```diff
@@ -50,11 +50,11 @@ var configTestCases = []configTestCase{
...
-			"iso-url":   "http://foo.bar/minishift-centos.iso",
-			"vm-driver": "kvm",
-			"cpus":      4,
-			"disk-size": "20g",
-			"v":         5,
+			"iso-url":                   "http://foo.bar/minishift-centos.iso",
+			"vm-driver":                 "kvm",
+			"cpus":                      4,
+			"disk-size":                 "20g",
+			"v":                         5,
```
Comment (praveenkumar) [inline cmd/minishift/cmd/config/config_test.go]: :+1: , I am not sure why that was not giving the error in case of fmt when using go1.10.x

**Comment by praveenkumar** [comment]
retest this please.

**Comment by gbraad** [comment]
centos ci misbehaving ... ?

**Comment by gbraad** [comment]
> Because we can not control the go verison in CentOS CI.

At the moment there is no modular or scl release of golang, although I overheard some discussions around this. We could if we use the bundled versions from golang.org if this really causes issues.

**Comment by praveenkumar** [comment]
> At the moment there is no modular or scl release of golang, although I overheard some discussions around this. We could if we use the bundled versions from golang.org if this really causes issues.

@gbraad right but our main aim for not using the bundled so we don't get any surprise when doing the downstream bits.

**Comment by gbraad** [comment]
it seems we get surprises anyway ;-)

**Comment by gbraad** [comment]
I believe part of this can be solved by utilizing a S2i image to perform the actual build. I have done this before... and I have alos seen OpenShift Origin do this for some time (Cesar Wong did so) for the `oc cluster` test releases.

File: cmd/minishift/cmd/config/config_test.go (hunk 2 above)
Comment (gbraad) [reply cmd/minishift/cmd/config/config_test.go]: that is a new thing... and I would even think a regression or a particular choice; the leader is a single space. maybe this is not distinct enough.

File: pkg/minishift/docker/image/oci_image_handler.go [hunk 3]
```diff
@@ -435,10 +435,10 @@ func (handler *OciImageHandler) getSystemContext(cacheDir string) *types.SystemC
...
 		DockerDaemonCertPath:              handler.dockerClientSettings.DockerCertPath,
 		DockerDaemonInsecureSkipTLSVerify: !handler.dockerClientSettings.DockerTLSVerify,
-		OSChoice:                    "linux",
-		ArchitectureChoice:          "amd64",
-		OCIAcceptUncompressedLayers: true,
-		OCISharedBlobDirPath:        filepath.Join(cacheDir, "blobs"),
+		OSChoice:                          "linux",
+		ArchitectureChoice:                "amd64",
+		OCIAcceptUncompressedLayers:       true,
+		OCISharedBlobDirPath:              filepath.Join(cacheDir, "blobs"),
```
Comment (gbraad) [inline pkg/minishift/docker/image/oci_image_handler.go]: a single space thing...

File: pkg/util/crypto.go [hunk 4]
```diff
@@ -62,7 +62,7 @@ func GenerateSelfSignedCert(certPath, keyPath string, ips []net.IP, alternateDNS
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
 		BasicConstraintsValid: true,
-		IsCA: true,
+		IsCA:                  true,
```
Comment (gbraad) [inline pkg/util/crypto.go]: weird this was allowed before... same in the previous file view_test.go.

File: pkg/util/os/atexit/atexit.go [hunk 5]
```diff
@@ -74,7 +74,7 @@ func runHandler(exitHandler func(code int) bool, code int) bool {
 	defer func() {
 		err := recover()
 		if err != nil {
-			glog.Exitf("Error running exit handler:", err)
+			glog.Exitf("Error running exit handler: %v", err)
```
Comment (gbraad) [inline pkg/util/os/atexit/atexit.go]: hmmm... they finally enforce/check formatting.

**Comment by gbraad** [comment]
I'll wait a little longer for CI... else I will go ahead and merge

**Comment by gbraad** [comment]
2 hours !

**Pull request merged**